src/pyspiffe/workloadapi/default_jwt_source.py

```diff
@@ -0,0 +1,150 @@
+                                 In case it's not provided, a default client will be created.
+            spiffe_socket_path: Path to Workload API UDS. This will be used in case a the workload_api_client is not provided.
+                           If not specified, the SPIFFE_ENDPOINT_SOCKET environment variable will be used and thus, must be set.
+            timeout_in_seconds: Time to wait for the first update of the Workload API. If not provided, and
+                                the connection with the Workload API fails, it will block indefinitely while
+                                the connection is retried.
+
+        Returns:
+            DefaultJwtSource: New DefaultJwtSource object, initialized with the JwtBundleSet fetched from the Workload API.
+
+        Raises:
+            ArgumentError: If spiffe_socket_path is invalid or not provided and SPIFFE_ENDPOINT_SOCKET env variable is not set.
+            JwtSourceError: In case a timeout was configured and it was reached during the source initialization waiting
+                             for the first update from the Workload API.
+        """
+
+        self._initialized = threading.Event()
+        self._lock = threading.Lock()
+        self._closed = False
+        self._workload_api_client = (
+            workload_api_client
+            if workload_api_client
+            else DefaultWorkloadApiClient(spiffe_socket_path)
+        )
+
+        # set the watcher that will keep the source updated and log the underlying errors
+        self._client_cancel_handler = self._workload_api_client.watch_jwt_bundles(
+            self._set_jwt_data, self._on_error
+        )
+        self._initialized.wait(timeout_in_seconds)
+
+        if not self._initialized.is_set():
+            self._client_cancel_handler.cancel()
+            raise JwtSourceError(
+                'Could not initialize JWT Source: reached timeout waiting for the first update'
+            )
+
+    def get_jwt_svid(
+        self, audiences: Set[str], subject: Optional[SpiffeId] = None
+    ) -> JwtSvid:
+        """Returns an JWT-SVID from the source.
+
+        Args:
+            audiences: List of audiences for the JWT SVID.
+            subject: SPIFFE ID subject for the JWT.
+
+        Raises:
+            ArgumentError: In case audiences is empty.
+            JwtSourceError: In case an error is raised when connecting to WORKLOAD API.
+        """
+        if not audiences:
+            raise ArgumentError('Audience cannot be empty')
+        try:
+            jwt_svid = self._workload_api_client.fetch_jwt_svid(audiences, subject)
+            return jwt_svid
+        except Exception as err:
+            logging.error('JWT Source: error getting JWT SVID: {}.'.format(str(err)))
+            logging.error('JWT Source: closing due to invalid state.')
+            self.close()
+            raise JwtSourceError(
+                'JWT Source: error getting JWT SVID: {}'.format(str(err))
+            )
```

Comment:
I'm wondering if any error should cause the source to be closed? 
`fetch_jwt_svid()` is a one-shot call that can return a validation error (like SPIFFE ID malformed or invalid) and in that case I think it might be reasonable to return an error without closing the source. Does it make sense?

If it does, we have another issue that is how to identify that kind of errors because `fetch_jwt_svid()` will raise `FetchJwtSvidError` in several cases. We need to think about this if necessary.

Reply:
Hi Andres, thanks! I agree... I have removed the close and removed the catch Exception :).